### PR TITLE
Projected gradients

### PR DESCRIPTION
**Pull request by MylesBartlett**

### CODE REVIEW

File: ethicml/implementations/pytorch_common.py [hunk 1]
```diff
@@ -3,11 +3,11 @@
 
 import numpy as np
 import torch
+import torch.nn as nn
+from ethicml.utility.data_structures import DataTuple, TestTuple
```
Comment (thomkeh) [inline ethicml/implementations/pytorch_common.py]: why is ethicml imported before third-party libraries?

File: ethicml/implementations/pytorch_common.py [hunk 2]
```diff
@@ -119,3 +119,18 @@ def pad_second(x):
...
     return mmd2
+
+
+def compute_projection_gradients(
+    model: nn.Module, loss_p: Tensor, loss_a: Tensor, alpha: float
+) -> None:
+    grad_p = torch.autograd.grad(loss_p, model.parameters(), retain_graph=True)
+    grad_a = torch.autograd.grad(loss_a, model.parameters(), retain_graph=True)
+
+    def _proj(a: Tensor, b: Tensor):
```
Comment (thomkeh) [inline ethicml/implementations/pytorch_common.py]: no jit?
